pkg/apis/experimental/types.go

```diff
@@ -508,35 +509,56 @@ type IngressStatus struct {
 	LoadBalancer api.LoadBalancerStatus `json:"loadBalancer,omitempty"`
 }
 
-// IngressRule represents the rules mapping the paths under a specified host to the related backend services.
-type IngressRule struct {
+// IngressHost represents the rules mapping the paths under a specified host to
+// the related backend services.
+type IngressHost struct {
 	// Host is the fully qualified domain name of a network host, or its IP
```

Comment:
Same as if this is hostname and != hostname on claim. Currently we'll program the loadbalancer with whatever you ask for, validating against the ip in status will only help so much, because you can still screw up hostname right? Though you could argue that we should invalidate what we can. 

Long run an ingress shouldn't be allowed to take another claim hostage. Cross resource validation?